src/content/en/updates/2018/05/bigint.md

```diff
@@ -0,0 +1,299 @@
+```
+
+Note how `max + 1` produces the same result as `max + 2`. Whenever we get this particular value in
+JavaScript, there is no way to tell whether it’s accurate or not. Any calculation on integers
+outside the safe integer range (i.e. from `Number.MIN_SAFE_INTEGER` to `Number.MAX_SAFE_INTEGER`)
+potentially loses precision. For this reason, we can only rely on numeric integer values within
+the safe range.
+
+## The new hotness: `BigInt` {: #bigint }
+
+`BigInt`s are a new numeric primitive in JavaScript that can represent integers with [arbitrary
+precision](https://en.wikipedia.org/wiki/Arbitrary-precision_arithmetic). With `BigInt`s, you can
+safely store and operate on large integers even beyond the safe integer limit for `Number`s.
+
+To create a `BigInt`, add the `n` suffix to any integer literal. For example, `123` becomes `123n`.
+The global `BigInt(number)` function can be used to convert a `Number` into a `BigInt`. In other
+words, `BigInt(123) === 123n`. Let’s use these two techniques to solve the problem we were having
+earlier:
+
+```js
+BigInt(Number.MAX_SAFE_INTEGER) + 2n;
+// → 9_007_199_254_740_993n ✅
+```
+
+Here’s another example, where we’re multiplying two `Number`s:
+
+```js
+1234567890123456789 * 123;
+// → 151851850485185200000 ❌
+```
+
+Looking at the least significant digits, `9` and `3`, we know that the result of the multiplication
+should end in `7` (because `9 * 3 === 27`). However, the result ends in a bunch of zeroes. That
+can’t be right! Let’s try again with `BigInt`s instead:
+
+```js
+1234567890123456789n * 123n;
+// → 151851850485185185047n ✅
+```
+
+This time we get the correct result.
+
+The safe integer limits for `Number`s don’t apply to `BigInt`s. Therefore, with `BigInt` we can
+perform correct integer arithmetic without having to worry about losing precision.
+
+### A new primitive
+
+`BigInt`s are a new primitive in the JavaScript language. As such, they get their own type that can
+be detected using the `typeof` operator:
+
+```js
+typeof 123;
+// → 'number'
+typeof 123n;
+// → 'bigint'
+```
+
+Because `BigInt`s are a separate type, a `BigInt` is never strictly equal to a `Number`, e.g.
+`42n !== 42`. To compare a `BigInt` to a `Number`, convert one of them into the other’s type before
+doing the comparison or use abstract equality (`==`):
+
+```js
+42n === BigInt(42);
+// → true
+42n == 42;
+// → true
+```
+
+When coerced into a boolean (which happens when using `if`, `&&`, `||`, or `Boolean(int)`, for
+example), `BigInt`s follow the same logic as `Number`s.
+
+```js
+if (0n) {
+  console.log('if');
+} else {
+  console.log('else');
+}
+// → logs 'else', because `0n` is falsy.
+```
+
+### Operators
+
+`BigInt`s support the most common operators. Binary `+`, `-`, `*`, and `**` all work as expected.
+`/` and `%` work, and round towards zero as needed. Bitwise operations `|`, `&`, <code>&lt;&lt;</code>,
+<code>&gt;&gt;</code>, and `^` perform bitwise arithmetic assuming a [two’s complement
```

Comment:
Had to use <code> here because `<<…>>` does a “Markdown include” on DevSite. 🤷🏼‍♂️